packages/storage-plus/src/indexed_map.rs

```diff
@@ -172,11 +172,11 @@ where
     K: PrimaryKey<'a>,
     I: IndexList<T>,
 {
-    pub fn sub_prefix(&self, p: K::SubPrefix) -> Prefix<K::SuperSuffix, T> {
+    pub fn sub_prefix(&self, p: K::SubPrefix) -> Prefix<K::SuperSuffix, T, K::SuperSuffix> {
```

Comment:
Ugh, so what is 3rd type argument in `Prefix` again?
```rust
pub struct Prefix<K, T, B = Vec<u8>>
where
    K: KeyDeserialize,
    T: Serialize + DeserializeOwned,
{
    storage_prefix: Vec<u8>,
    data: PhantomData<(T, B)>,
    pk_name: Vec<u8>,
    de_fn_kv: DeserializeKvFn<K, T>,
    de_fn_v: DeserializeVFn<T>,
}
```
It's the actual content of prefix, right?

Reply:
The 3rd type is precisely the new `Bound` type annotations. You can compare with `Map` to get the idea.

While implementing this, made it work for `Map`, and then forgot about these here and elsewhere (`MultiIndex`). These bounds have a default of `Vec<u8>`, that's why it "worked".

Adding explicit tests for these typed bounds would be good I guess, at some point.

Reply:
It's a little tricky how all this works. I can give a small overview / talk at some point, if there's interest.